File: app/src/main/java/com/snail/labaffinity/app/MyButton.java
package com.snail.labaffinity.app;

import android.content.Context;
import android.graphics.Canvas;
import android.os.SystemClock;
import android.util.AttributeSet;

import com.snail.labaffinity.utils.LogUtils;

public class MyButton extends androidx.appcompat.widget.AppCompatTextView {
    public MyButton(Context context) {
        super(context);
    }

    public MyButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MyButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        SystemClock.sleep(1000);
        LogUtils.v("onDraw");
    }

    @Override
    protected void onAttachedToWindow() {
        super.onAttachedToWindow();
        LogUtils.v("onAttachedToWindow");
    }

    @Override
    public void onWindowFocusChanged(boolean hasWindowFocus) {
        super.onWindowFocusChanged(hasWindowFocus);
        LogUtils.v("onWindowFocusChanged");
    }
}

File: app/src/main/java/com/snail/labaffinity/app/MyButton.kt
package com.snail.labaffinity.app

import android.content.Context
import android.graphics.Canvas
import android.os.SystemClock
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.snail.labaffinity.utils.LogUtils

class MyButton constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) :
    AppCompatTextView(
        context,
        attrs,
        defStyleAttr
    ) {
    constructor(context: Context?) : this(context, null) {}
    constructor(context: Context?, attrs: AttributeSet?) : this(context, attrs, 0) {}


    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        SystemClock.sleep(1000)
        LogUtils.v("onDraw")
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        LogUtils.v("onAttachedToWindow")
    }

    override fun onWindowFocusChanged(hasWindowFocus: Boolean) {
        super.onWindowFocusChanged(hasWindowFocus)
        LogUtils.v("onWindowFocusChanged")
    }
}